etcdserver/api/v3rpc/auth.go

```diff
@@ -68,8 +68,7 @@ func (as *AuthServer) RoleGrant(ctx context.Context, r *pb.RoleGrantRequest) (*p
 }
 
 func (as *AuthServer) UserAdd(ctx context.Context, r *pb.UserAddRequest) (*pb.UserAddResponse, error) {
-	plog.Info("not implemented yet")
-	return nil, nil
+	return as.authenticator.UserAdd(ctx, r)
```

Comment:
add the duplicate error and exhausted error into v3rpc/errors.go. then we can use gRPC to return the error.